clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -84,6 +138,19 @@ mlir::LogicalResult CIRToLLVMGlobalOpLowering::matchAndRewrite(
   SmallVector<mlir::NamedAttribute> attributes;
 
   if (init.has_value()) {
+    auto setupRegionInitializedLLVMGlobalOp = [&]() {
```

Comment:
I think I'd prefer this get extracted to a function in an anonymous namespace (or a static function).  It does quite a lot of work for a lambda.

Reply:
It was a separate function in the incubator. The reason I moved it to a lambda is that there was a lot of state information needed that would either need to be passed in as arguments or recreated. The state is per-global so it can't just be stored in the CIRToLLVMGlobalOpLowering object. It's the set of arguments that are passed to replaceOpWithNewOp()

Reply:
Oh! I see, I missed how much it was capturing.  Can you link me to the incubator version?

Reply:
https://github.com/llvm/clangir/blob/f8821e8521686749ab43015cd7a5b2f863b9943d/clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp#L2344

Reply:
It's not quite as bad there because everything is coming from the `op` object. It was worse here because of all the placeholders for things the `op` doesn't have yet here. Still the incubator implementation retrieves a lot of this in multiple places and only calls the function in this scope.

Reply:
Ah, hrmph.  That is unfortunately a much nicer version as a separate function.  Can we have a FIXME to move this once dataLayout, typeConverter, and init.value get moved? (or whatever else needs to be moved?).  It would be great if we could do that sooner, but I see what you mean.